app/controllers/ideas_controller.rb

```diff
@@ -9,20 +9,25 @@ def new
   def create
     @user = User.find(params[:user_id])
     @idea = @user.ideas.new(idea_params)
-      if @idea.save
-        redirect_to user_idea_path(@user, @idea)
-      else
-        render :new
-      end
+    if @idea.save
+      redirect_to user_idea_path(@user, @idea)
+    else
+      render :new
+    end
   end
 
   def index
     @ideas = Idea.all
   end
 
   def show
-    @idea = Idea.find(params[:id])
-    @user = @idea.user
+    # byebug
+    if current_user.id == params[:user_id].to_i
```

Comment:
Here its hitting an undefined method id as nil. I've checked current user here and I get 
`<ActionController::Parameters {"controller"=>"ideas", "action"=>"show", "user_id"=>"7", "id"=>"1"} permitted: false>` as params

Reply:
so I would think that since current_user returns some kind of ID, tis not nil